src/hacbs/components/ApplicationDetails/DetailsPage.tsx

```diff
@@ -0,0 +1,151 @@
+import * as React from 'react';
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownItemProps,
+  DropdownSeparator,
+  DropdownToggle,
+  Flex,
+  FlexItem,
+  PageGroup,
+  PageSection,
+  PageSectionVariants,
+  Tab,
+  Tabs,
+  TabTitleText,
+  Text,
+  TextContent,
+} from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons/dist/esm/icons/caret-down-icon';
+import BreadCrumbs from '../../../shared/components/breadcrumbs/BreadCrumbs';
+import { getQueryArgument, removeQueryArguments, setQueryArgument } from '../../../shared/utils';
+
+type Action = { type?: string; key: string; label: string } & DropdownItemProps;
+type TabProps = {
+  key: string;
+  label: string;
+  component: React.ReactNode;
+  href?: string;
+  isDisabled?: true;
+};
+
+type DetailsPageProps = {
+  title: string;
+  children?: React.ReactNode;
+  footer?: React.ReactNode;
+  description?: React.ReactNode;
+  breadcrumbs?: { name: string; path: string }[];
+  actions?: Action[];
+  tabs: TabProps[];
+  onTabSelect?: (selectedTabKey: string) => void;
+};
+
+const DetailsPage: React.FC<DetailsPageProps> = ({
+  title,
+  footer,
+  description,
+  breadcrumbs,
+  actions = [],
+  tabs = [],
+  onTabSelect,
+}) => {
+  const tabMatched =
+    tabs?.find((t) => t.key === getQueryArgument('activeTab'))?.key || tabs?.[0]?.key;
```

Comment:
You should be reading from react router for query changes to ensure reactivity. Right now the re-render depends on a higher up component causing this child to be re-rendered. But it's possible that this could break by using a pure component somewhere in between.

That being said, I'm not sure having the active tab as a query param is correct. For now it works though. We should talk with @rohitkrai03 to align on when to use query params vs path segments as we also use query params for the resource name.